src/main/java/com/google/sps/servlets/RandomRecipeServlet.java

```diff
@@ -57,47 +60,53 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
    * This function generates a random double and returns the id of the first Entity with the closest value of "random_number".
    */
   private long returnRandomId(DatastoreService datastore) {
-    Random randomGenerator = new Random();
-    Double numberGenerated = randomGenerator.nextDouble();
-    try {
-      return closestRecipeRandomId(
-        numberGenerated,
-        datastore,
-        Query.FilterOperator.LESS_THAN_OR_EQUAL,
-        Query.SortDirection.DESCENDING
-      );
-    } catch (IndexOutOfBoundsException e) {
-      return closestRecipeRandomId(
-        numberGenerated,
-        datastore,
-        Query.FilterOperator.GREATER_THAN_OR_EQUAL,
-        Query.SortDirection.ASCENDING
-      );
+    int position = getRandomPosition(datastore);
+    Results<ScoredDocument> recipes = getRecipeDocumentFromPosition(
+      datastore,
+      position
+    );
+    for (ScoredDocument recipe : recipes) {
+      return Long.parseLong(recipe.getId());
```

Comment:
Why you don't just get the first one instead of doing this?

Reply:
I haven't found a way to get the first element from "Results" array. I'll search more.